src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs

```diff
@@ -1300,12 +1300,20 @@ private ParameterMetadata RehydrateParameterMetadata(PSObject deserializedParame
                 parameterType);
         }
 
-        private static bool IsProxyForCmdlet(Dictionary<string, ParameterMetadata> parameters)
+        private bool IsProxyForCmdlet(Dictionary<string, ParameterMetadata> parameters)
         {
             // we are not sending CmdletBinding/DefaultParameterSet over the wire anymore
             // we need to infer IsProxyForCmdlet from presence of all common parameters
 
-            foreach (string commonParameterName in Cmdlet.CommonParameters)
+            var commonParameters = new HashSet<string>(Cmdlet.CommonParameters);
```

Comment:
Instead of always creating a copy when reaching here, how about changing the code to something like this:
```c#
bool isDownLevelRemote = Session.Runspace is RemoteRunspace remoteRunspace
    && remoteRunspace.ServerVersion is not null
    && remoteRunspace.ServerVersion <= new Version(7, 3);

foreach (string commonParameterName in CommonParameters)
{
    if (isDownLevelRemote && commonParameterName == "ProgressAction")
    {
        continue;
    }

    if (!parameters.ContainsKey(commonParameterName))
    {
        return false;
    }
}
```